### ISSUE TITLE
Try 2: Remove deprecated inheritance of SerialDense matrices from Object

### ISSUE THREAD

**Issue by rhoope** [trilinos/Trilinos#4430]
This is a new issue for the idea captured in Issue #4258 but using an approach that doesn't break backward compatibility (cf Issue #4330) and attempts to align with current guidance for deprecating code as in Issue #4269).

@trilinos/teuchos

## Motivation and Context

Details are provided in Issue #4258.
## Your Environment

- **Relevant repo SHA1s:**  
- **Relevant configure flags or configure script:**  
- **Operating system and version:**  
- **Compiler and TPL versions:**  

## Related Issues

* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to Issues #4258 and #4330
* Part of 
* Composed of Issue #4258

**Comment by mhoemmen** [comment]
@trilinos/teuchos

**Comment by rhoope** [comment]
Thanks much @mhoemmen !

**Comment by mhoemmen** [comment]
@rhoope Thanks for all your hard work! :D
